### PR TITLE
Initial doc on HTTP stats detailing just client metrics.

### PR DESCRIPTION
**Pull request by ramonza**

### CODE REVIEW

File: http/stats.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+| headers_recv       | none                     | Total number of header lines received in responses, not including trailing headers                                                                                                                                                                                                                |
+| latency            | distribution             | Time between first byte of request headers sent to last byte of response received, or terminal error                                                                                                                                                                                              |
+| ttfb               | distribution             | Time between last byte of request sent and first byte of response body received                                                                                                                                                                                                                   |
```
Comment (bogdandrutu) [inline http/stats.md]: This is very hard to measure correctly, I would remove it or put an optional for this.

File: http/stats.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+### Measures
+
+Client stats are recorded for each individual HTTP request, including for each redirect (followed or not). 
+
+| Measure suffix     | Default View Aggregation | Description                                                                                                                                                                                                                                                                                       |
+|--------------------|--------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| started            | sum                      | Number of all client requests started                                                                                                                                                                                                                                                             |
```
Comment (bogdandrutu) [inline http/stats.md]: I would not add this for the moment, or maybe just mark it as optional.

File: http/stats.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+| latency            | distribution             | Time between first byte of request headers sent to last byte of response received, or terminal error                                                                                                                                                                                              |
+| ttfb               | distribution             | Time between last byte of request sent and first byte of response body received                                                                                                                                                                                                                   |
+| connections_opened | count                    | Number of underlying transport (TCP/TLS) connections opened                                                                                                                                                                                                                                       |
```
Comment (bogdandrutu) [inline http/stats.md]: Is this the total number of connection or is the active connections?

File: http/stats.md (hunk 2 above)
Comment (ramonza) [reply http/stats.md]: The way I see it, everything is optional. I meant this as more like if the language provides this metric then this is what it's name should be. Curious why you think this is not an important metric?

File: http/stats.md (hunk 1 above)
Comment (ramonza) [reply http/stats.md]: Go has a good way to measure this (httptrace) and often it's a useful metric. I will comment at the top of the file that not all libraries provide all stats.

File: http/stats.md (hunk 3 above)
Comment (ramonza) [reply http/stats.md]: I imagined it was just the total number opened. I can see the latter being useful as well so I'll add a connections_closed as well.

Comment (bogdandrutu) [reply http/stats.md]: Sounds good.

File: http/stats.md (hunk 1 above)
Comment (bogdandrutu) [reply http/stats.md]: Please comment.

File: http/stats.md (hunk 2 above)
Comment (bogdandrutu) [reply http/stats.md]: I did not want to say that it is not important, in general we get this information from the tracing side by looking at the number of active HTTP spans.

Happy to keep it as optional (as you mentioned all the metrics are optional).

FYI: also in general for the number of completed requests we get that from the latency distribution by querying the total count you can mentioned that in this doc.

File: http/stats.md (hunk 1 above)
Comment (ramonza) [reply http/stats.md]: Actually after looking into this further, I think I agree with you that it's not that useful and has definitional problems. I will remove it.

File: http/stats.md (hunk 2 above)
Comment (ramonza) [reply http/stats.md]: Yes, this is how the view "ended" is defined below.

**Pull request merged**